### PR TITLE
Debug cart

### PR DESCRIPTION
**Pull request by 0Rawan**
#83

### CODE REVIEW

File: src/controllers/cart.js [hunk 1]
```diff
@@ -74,41 +75,38 @@ exports.putCartItemByDishId = async (req, res, next) => {
...
       'cartItems.dishId': req.params.id,
     });
-
-    if (cart.cartItems[0]) {
+    if (cart) {
       cart.cartItems.splice(0, 1);
       cart.save();
-    } else {
-      res.status(404).json({ message: 'item not found' });
+      return res.status(204).json(cart);
```
Comment (HishamWattar) [inline src/controllers/cart.js:106]: why did you remove ```catch (err) {
    return next(new CustomError(err.message, 422));```
if the time was not exists you should an error message

File: src/controllers/cart.js [hunk 2]
```diff
@@ -19,18 +20,18 @@ exports.postCart = async (req, res, next) => {
...
+    return next(new CustomError(err.message));
   }
 };
 
-exports.deleteCart = async (req, res) => {
+exports.deleteCart = async (req, res, next) => {
   try {
     Cart.deleteOne({ customerId: req.user.id });
-    res.status(204).json({
+    return res.status(204).json({
```
Comment (HishamWattar) [inline src/controllers/cart.js:30]: If the status code is 204, it means nothing will be displayed in the response body
so instead use this: ```return res.sendStatus(204);```

File: logs/combined.log
Comment (HishamWattar) [inline logs/combined.log:1]: this file should not be included

File: logs/error.log
Comment (HishamWattar) [inline logs/error.log:1]: this file should not be included

File: logs/exceptions.log
Comment (HishamWattar) [inline logs/exceptions.log:1]: this file should not be included

File: src/controllers/cart.js (hunk 1 above)
Comment (0Rawan) [reply src/controllers/cart.js:106]: it's in line 108, I did code refactoring

**Review by HishamWattar — APPROVED** [review]

**Review by ahmadalashtar — APPROVED** [review]
